src/Bicep.Core/Semantics/Namespaces/SystemNamespaceSymbol.cs

```diff
@@ -389,6 +391,67 @@ public class SystemNamespaceSymbol : NamespaceSymbol
+                .WithRequiredParameter("filePath", LanguageConstants.String, "File path which content will be loaded")
+                .WithOptionalParameter("encoding", LanguageConstants.String, "File encoding. If not provided, Bicep will try to automatically detect the encoding of a file based on the presence of byte order marks.")
+                .WithDiagnosticsCollector((binder, diagnostics, arguments) => {
+                    if (arguments[0].syntax.Expression is StringSyntax filePathSyntax)
+                    {
+                        if (filePathSyntax.IsInterpolated())
+                        {
+                            diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[0].syntax).CompileTimeConstantRequired());
+                            return;
+                        }
+
+                        var filePath = filePathSyntax.TryGetLiteralValue() ?? string.Empty;
+                        if (!SyntaxTreeGroupingBuilder.ValidateFilePath(filePath, out var validateFilePathFailureBuilder))
+                        {
+                            diagnostics.Write(validateFilePathFailureBuilder.Invoke(DiagnosticBuilder.ForPosition(arguments[0].syntax)));
+                            return;
+                        }
+
+                        var fileUri = binder.FileResolver.TryResolveModulePath(binder.FileUri, filePath);
+                        if (fileUri is null)
+                        {
+                            diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[0].syntax).FilePathCouldNotBeResolved(filePath, binder.FileUri.LocalPath));
+                            return;
+                        }
+
+                        if (!fileUri.IsFile)
+                        {
+                            diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[0].syntax).UnableToLoadNonFileUri(fileUri));
+                            return;
+                        }
+
+
+                        Encoding? fileEncoding = null;
+                        if (arguments.Length > 1)
+                        {
+                            fileEncoding = Encoding.UTF8; //todo read encoding
+                        }                        
+
+                        if (!binder.FileResolver.TryRead(fileUri, out var fileContent, out var fileReadFailureBuilder, fileEncoding))
+                        {
+                            diagnostics.Write(fileReadFailureBuilder.Invoke(DiagnosticBuilder.ForPosition(arguments[0].syntax)));
+                            return;
+                        }
+
+                        if (fileContent.Length > 24576)
+                        {
+                            diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[0].syntax).FileExceedsMaximumSize(fileUri.LocalPath, 24576, "characters"));
+                        }
+                    }
+                })
+                .Build(),
+
+            new FunctionOverloadBuilder("loadFile")
```

Comment:
>loadFile [](start = 41, length = 8)

The distinction is really subtle in behavior between `loadFile` and `loadContent`. @alex-frankel wanted to use parameters for this, so I think we should stick to that. (I think I also may have misled you with my comment on the signature. It was using `loadFile` as an example function name.)

Reply:
I thought about a parameter too, but I wanted to have a parameter with file's encoding used to read as text. But perhaps in addition to 'auto' and 'utf-8' 'ascii' or other - we can have an enum value 'binary' that would load as base64.
I'll merge them.

Reply:
You are going too fast! We are talking about this right now :)

We are leaning towards *keeping* the two functions, but clarifying the names:

`loadBinaryContent()`
`loadTextContent()`

@anthony-c-martin is going to share more details on what the function signatures should look like.

Reply:
Sure, just wanted to start with some ground work, mostly how and when to actually read the file :) I feel like this is only 10% of the final solution, so lots of work ahead :)